validator/accounts/v2/new.go

```diff
@@ -0,0 +1,265 @@
+	case v2keymanager.Direct:
+		keymanager, err = direct.NewKeymanagerFromConfigFile(ctx, wallet)
+		if err != nil {
+			log.Fatal(err)
+		}
+	case v2keymanager.Derived:
+		log.Fatal("Derived keymanagers are unimplemented, work in progress")
+	case v2keymanager.Remote:
+		log.Fatal("Remote keymanagers are unimplemented, work in progress")
+	default:
+		log.Fatal("Keymanager kind must be specified")
+	}
+	return keymanager
+}
+
+// Check if a user has an existing wallet at the specified path.
+func hasWalletDir(walletPath string) (bool, error) {
+	_, err := os.Stat(walletPath)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return true, err
+}
+
+func inputWalletDir(cliCtx *cli.Context) string {
+	datadir := cliCtx.String(flags.WalletDirFlag.Name)
+	prompt := promptui.Prompt{
+		Label:    "Enter a wallet directory",
+		Validate: validateDirectoryPath,
+		Default:  datadir,
+	}
+	walletPath, err := prompt.Run()
+	if err != nil {
+		log.Fatalf("Could not determine wallet directory: %v", formatPromptError(err))
+	}
+	return walletPath
+}
+
+func inputKeymanagerKind(_ *cli.Context) v2keymanager.Kind {
+	promptSelect := promptui.Select{
+		Label: "Select a type of wallet",
+		Items: []string{
+			keymanagerKindSelections[v2keymanager.Direct],
+			keymanagerKindSelections[v2keymanager.Derived],
+			keymanagerKindSelections[v2keymanager.Remote],
+		},
+	}
+	selection, _, err := promptSelect.Run()
+	if err != nil {
+		log.Fatalf("Could not select wallet type: %v", formatPromptError(err))
+	}
+	return v2keymanager.Kind(selection)
+}
+
+func inputAccountPassword(_ *cli.Context) string {
+	var hasValidPassword bool
+	var walletPassword string
+	for !hasValidPassword {
+		prompt := promptui.Prompt{
+			Label:    "New account password",
+			Validate: validatePasswordInput,
+			Mask:     '*',
+		}
+
+		walletPassword, err := prompt.Run()
+		if err != nil {
+			log.Fatalf("Could not read wallet password: %v", formatPromptError(err))
+		}
+
+		prompt = promptui.Prompt{
+			Label: "Confirm password",
+			Mask:  '*',
+		}
+		confirmPassword, err := prompt.Run()
+		if err != nil {
+			log.Fatalf("Could not read password confirmation: %v", formatPromptError(err))
+		}
+		if walletPassword != confirmPassword {
+			log.Error("Passwords do not match")
+			continue
+		}
+		hasValidPassword = true
+	}
+	return walletPassword
+}
+
+func inputPasswordsDirectory(cliCtx *cli.Context) string {
+	passwordsDir := cliCtx.String(flags.WalletPasswordsDirFlag.Name)
+	prompt := promptui.Prompt{
+		Label:    "Passwords directory",
+		Validate: validateDirectoryPath,
+		Default:  passwordsDir,
+	}
+	passwordsPath, err := prompt.Run()
+	if err != nil {
+		log.Fatalf("Could not determine passwords directory: %v", formatPromptError(err))
+	}
+	return passwordsPath
+}
+
+// Validate a strong password input for new accounts,
+// including a min length, at least 1 number and at least
+// 1 special character.
+func validatePasswordInput(input string) error {
```

Comment:
consider using this: https://github.com/nbutton23/zxcvbn-go

to password such as `aaaaaa1!`

Reply:
Added